dao/src/main/java/com/epam/esm/dao/CertificateDao.java

```diff
@@ -0,0 +1,63 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.dao.specification.Specification;
+import com.epam.esm.entity.Certificate;
+
+import java.util.List;
+import java.util.Optional;
+
+/** The interface Certificate dao. */
+public interface CertificateDao {
+  /**
+   * Find certificates by specification.
+   *
+   * @return A list of certificates
+   */
+  List<Certificate> findBySpecification(Specification specification, String sortParams);
+
+  /**
+   * Find by certificate id.
+   *
+   * @param id the certificate id
+   * @return The founded certificate
+   */
+  Optional<Certificate> findById(long id);
+
+  /**
+   * Create certificate.
+   *
+   * @param certificate the certificate to be placed in the database
+   * @return The created certificate
+   */
+  Optional<Certificate> save(Certificate certificate);
```

Comment:
What is the purpose of Optional here? Is it possible to get an empty optional? If yes, how to know what went wrong?
The same for update.